nucleus/common/jul-extensions/src/main/java/fish/payara/jul/PayaraLogger.java

```diff
@@ -0,0 +1,524 @@
+        }
+        checkAndLog(record);
+    }
+
+
+    /**
+     * {@inheritDoc}
+     * <p>
+     * This method is <b>overriden</b> because it is called from clients to check if it makes
+     * sense to do some expensive preparation and call of any log method, but we don't
+     * know it yet, so we accept everything to queue (it should not be so many records).
+     */
+    @Override
+    public boolean isLoggable(final Level level) {
+        return isLoggable(level, getLoggingStatus());
+    }
+
+
+    /**
+     * @param level a message logging level
+     * @param status the state of the logging system
+     * @return true if the given message level is currently being logged or if we don't know yet.
+     */
+    protected boolean isLoggable(final Level level, final PayaraLoggingStatus status) {
+        // if it is to queue, we cannot decide it for now, but
+        // we will check again later when processing the queue
+        return isLoggableLevel(level) || isToQueue(status);
+    }
+
+
+    /**
+     * @param level - log record level
+     * @return true if level has higher importance than level set to this logger (or parent)
+     */
+    protected boolean isLoggableLevel(final Level level) {
+        return super.isLoggable(level);
+    }
+
+
+    private boolean isReady(final PayaraLoggingStatus status) {
+        return status == PayaraLoggingStatus.FULL_SERVICE || status == PayaraLoggingStatus.FLUSHING_BUFFERS;
+    }
+
+
+    private boolean isToQueue(final PayaraLoggingStatus status) {
+        return status == PayaraLoggingStatus.UNCONFIGURED || status == PayaraLoggingStatus.CONFIGURING;
+    }
+
+
+    void checkAndLog(final LogRecord record) {
+        if (!isLoggableLevel(record.getLevel())) {
+            return;
+        }
+        final Filter filter = getFilter();
+        if (filter != null && !filter.isLoggable(record)) {
+            return;
+        }
+
+        final Predicate<Handler> handlerFilter = h -> h.isLoggable(record);
+        final List<Handler> handlers = getLoggers().stream().flatMap(LOGER_TO_HANDLER).filter(handlerFilter)
+            .collect(Collectors.toList());
+        if (handlers.isEmpty()) {
+            return;
+        }
+        final EnhancedLogRecord resolvedLogRecord = MSG_RESOLVER.resolve(record);
+        handlers.forEach(handler -> handler.publish(resolvedLogRecord));
+    }
+
+
+    private List<Logger> getLoggers() {
+        final ArrayList<Logger> loggers = new ArrayList<>();
+        Logger log = this;
+        do {
+            loggers.add(log);
+            final boolean useParentHandlers = log.getUseParentHandlers();
+            if (!useParentHandlers) {
+                break;
+            }
+            log = log.getParent();
+        } while (log != null);
+        return loggers;
+    }
+
+
+    private PayaraLoggingStatus getLoggingStatus() {
+        if (PayaraLogManager.isPayaraLogManager()) {
+            PayaraLogManager.getLogManager().getLoggingStatus();
```

Comment:
Missing Return statement?

Reply:
I am looking at it too, right now, I'm writing test for this. Seems I committed some unfinished work some time back and as it is not covered by tests ... I am happy you found it :-)